drivers/timer/nrf_rtc_timer.c

```diff
@@ -301,9 +292,7 @@ static int compare_set_nolocks(int32_t chan, uint64_t target_time,
 			/* Target time is valid and is different than currently set.
 			 * Set CC value.
 			 */
-			uint32_t cc_set = set_absolute_alarm(chan, cc_value);
-
-			target_time += counter_sub(cc_set, cc_value);
+			set_absolute_alarm(chan, cc_value);
```

Comment:
Yes, it was intentional. `target_time` is checked in `process_channel()`, so if it was updated here, the channel handler execution would always need to wait for the event generated by the last CC setting done by `set_absolute_alarm()`. Without the update, it can be executed on the first event generated for the alarm.